multifm/demod.c

```diff
@@ -293,6 +294,9 @@ aresult_t demod_thread_new(struct demod_thread **pthr, unsigned core_id,
     thr->fifo_fd = -1;
     thr->debug_signal_fd = -1;
 
+    // Pass in the CSQ variable
+    thr->csq_level_dbfs = csq_level_dbfs;
```

Comment:
It might be worthwhile to have a flag that indicates whether or not CSQ has been enabled, so we can selectively enable the code path that performs the per-buffer checks.